community/cypher/cypher/src/test/scala/org/neo4j/cypher/EagerizationAcceptanceTest.scala

```diff
@@ -1362,21 +1365,49 @@ class EagerizationAcceptanceTest extends ExecutionEngineFunSuite with TableDrive
     createNode()
     createNode("prop" -> 42)
     createNode("prop" -> 42)
-    val query = "MATCH (n) CREATE (m) WITH * MATCH (o {prop:42}) SET n.prop = 42 RETURN count(*)"
+    val query = "MATCH (n) CREATE (m) WITH * MATCH (o {prop:42}) SET n.prop = 42 RETURN count(*) as count"
 
     val result = updateWithBothPlanners(query)
+    result.columnAs[Int]("count").next should equal(8)
     assertStats(result, propertiesSet = 8, nodesCreated = 4)
     assertNumberOfEagerness(query, 1)
   }
 
+  test("setting property in tail should be eager if overlap head") {
+    createNode()
+    createNode()
+    createNode("prop" -> 42)
+    createNode("prop" -> 42)
+    val query = "MATCH (n {prop: 42}) CREATE (m) WITH * MATCH (o) SET n.prop = 42 RETURN count(*) as count"
+
+    val result = updateWithBothPlanners(query)
+    result.columnAs[Int]("count").next should equal(12)
+    assertStats(result, propertiesSet = 12, nodesCreated = 2)
+    assertNumberOfEagerness(query, 2)
+  }
+
+  test("setting property in tail should be eager if overlap first tail") {
+    createNode()
+    createNode()
+    createNode("prop" -> 42)
+    createNode("prop" -> 42)
+    val query = "CREATE () WITH * MATCH (n {prop: 42}) CREATE (m) WITH * MATCH (o) SET n.prop = 42 RETURN count(*) as count"
+
+    val result = updateWithBothPlanners(query)
+    result.columnAs[Int]("count").next should equal(14)
+    assertStats(result, propertiesSet = 14, nodesCreated = 3)
+    assertNumberOfEagerness(query, 3)
```

Comment:
I think this should just need 2 eagerness if we were smarter with set property overlap.. (the top repeatable read should be unnecessary)